src/Collection/Collection.router.ts

```diff
@@ -241,70 +246,77 @@ export class CollectionRouter extends Router {
 
   publishCollection = async (
     req: AuthRequest
-  ): Promise<{ collection: FullCollection; items: FullItem[] }> => {
+  ): Promise<{
+    collection: FullCollection
+    items: FullItem[]
+    itemCurations?: ItemCurationAttributes[]
+  }> => {
```

Comment:
It might make sense to type this with a `type` of its own? cause is being used [here](https://github.com/decentraland/builder-server/blob/815462f6592d138ebd62aa946f355bb18ce430ab/src/Collection/Collection.router.ts#L249), [on the result definition](https://github.com/decentraland/builder-server/blob/815462f6592d138ebd62aa946f355bb18ce430ab/src/Collection/Collection.router.ts#L259), [and on the service](https://github.com/decentraland/builder-server/blob/815462f6592d138ebd62aa946f355bb18ce430ab/src/Collection/Collection.router.ts#L259)

it could be used in `publishDCLCollection` too.

Reply:
Good idea, I added `PublishCollectionResponse` with a generic for the type of collection to return. 
`publishCollection` returns a `FullCollection` but `this.service.publishTPCollection` returns `CollectionAttributes` for the `collections` attribute so I had to add a generic to the type to support both (mainly because they are not compatible)